### PR TITLE
store: add retry block for bucket initial sync

### LINKED ISSUE #4489: bucket store initial sync: sync block: BaseFetcher: iter bucket: Server not initialized, please try again.
Store initial sync fails on temporary bucket error without retrying.

Bucket: S3.

Repro: I started minio and immediately then Thanos Store and it failed because minio was still starting (despite readiness probe being OK BTW :/ )

### PR DESCRIPTION
**Pull request by someshkoli**
* [ ] I added CHANGELOG entry for this change.
* [X] Change is not relevant to the end user.

## Changes
Pushes bucket initial sync into a retry block to avoid server shutdown due to objstore unavailability. 
Closes #4489

### CODE REVIEW

File: cmd/thanos/store.go [hunk 1]
```diff
@@ -361,11 +361,19 @@ func runStore(
 			defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
 
 			level.Info(logger).Log("msg", "initializing bucket store")
-			begin := time.Now()
-			if err := bs.InitialSync(ctx); err != nil {
-				close(bucketStoreReady)
-				return errors.Wrap(err, "bucket store initial sync")
+			// note(someshkoli): flag required to input interval duration ?
+			if err := runutil.RetryWithLog(logger, time.Second*2, ctx.Done(), func() error {
```
Comment (bwplotka) [inline cmd/thanos/store.go:365]: I would still fail, after some time. Can we set timeout for this? e.g 5m?

File: cmd/thanos/store.go [hunk 2]
```diff
@@ -361,11 +361,19 @@ func runStore(
...
+				if err := bs.InitialSync(ctx); err != nil {
+					return errors.Wrap(err, "bucket store initial sync")
+				}
+				return nil
+			}); err != nil {
+				runutil.CloseWithLogOnErr(logger, bs, "bucket store")
+				return err
 			}
+
+			begin := time.Now()
```
Comment (bwplotka) [inline cmd/thanos/store.go:375]: This does not make sense (:

Comment (bwplotka) [reply cmd/thanos/store.go:375]: let's add it above

**Review by bwplotka — COMMENTED** [review]
Nice, thanks! Some suggestions

**Comment by someshkoli** [comment]
Still valid

**Pull request closed without merging**